### PR TITLE
api: repos/search: add parameters to control the sort order

### PR DESCRIPTION
**Pull request by aswild**
The backend SearchRepositoryByName function supports several sort order
options, hook these up to the /repos/search API.

New parameters for /repos/search:
    'sort':  alpha, created, updated, size, or id
    'order': 0 for ascending or 1 for descending

'order' defaults to ascending, and is only used when 'sort' is also specified.
The overall default behavior is unchanged (alphabetically ascending).

This partially implements issue #3963

Signed-off-by: Allen Wild <allenwild93@gmail.com>

### CODE REVIEW

**Comment by lunny** [comment]
CI restarted

**Review by lafriks — APPROVED** [review]

File: routers/api/v1/repo/repo.go [hunk 1]
```diff
@@ -88,6 +118,22 @@ func Search(ctx *context.APIContext) {
 		return
 	}
 
+	var sortMode = ctx.Query("sort")
+	if len(sortMode) > 0 {
+		var sortOrder = ctx.QueryInt("order") // will default to 0, i.e. ascending
```
Comment (lafriks) [inline routers/api/v1/repo/repo.go]: It would be better if order would be string (`asc`/`desc`)

**Review by lafriks — CHANGES_REQUESTED** [review]

File: routers/api/v1/repo/repo.go (hunk 1 above)
Comment (aswild) [reply routers/api/v1/repo/repo.go]: Fixed. asc/desc does make more sense than 0/1

File: routers/api/v1/repo/repo.go [hunk 2]
```diff
@@ -20,6 +20,25 @@ import (
 	api "code.gitea.io/sdk/gitea"
 )
 
+var searchOrderByMap = [2]map[string]models.SearchOrderBy{
```
[the remark this answers is not in the document]
Comment (aswild) [reply routers/api/v1/repo/repo.go]: Good idea, I'll do that.

**Review by axifive — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Review by appleboy — APPROVED** [review]

**Pull request merged**